rest/tests/base.py

```diff
@@ -0,0 +1,389 @@
+"""
+LORE test case
+"""
+
+from __future__ import unicode_literals
+import json
+
+from rest_framework.status import (
+    HTTP_200_OK,
+    HTTP_201_CREATED,
+    HTTP_204_NO_CONTENT,
+)
+from rest_framework.reverse import reverse
+
+from learningresources.tests.base import LoreTestCase
+
+API_BASE = '/api/v1/'
+REPO_BASE = '/api/v1/repositories/'
+
+
+def as_json(resp):
+    """Get JSON from response"""
+    return json.loads(resp.content.decode('utf-8'))
+
+
+# pylint: disable=too-many-public-methods, too-many-arguments
+class RESTTestCase(LoreTestCase):
+    """
+    REST test case
+    """
+
+    # pylint: disable=dangerous-default-value
+    # Do not mutate these fields!
+    DEFAULT_REPO_DICT = {
+        'name': 'name',
+        'description': 'description',
+    }
+    DEFAULT_VOCAB_DICT = {
+        'name': 'name',
+        'description': 'description',
+        'required': True,
+        'weight': 1000,
+        'vocabulary_type': 'f',
+    }
+    DEFAULT_TERM_DICT = {
+        'label': 'term label',
+        'weight': 1000,
+    }
+
+    def assert_options_head(self, url, expected_status):
+        """Assert OPTIONS and HEAD"""
+        resp = self.client.options(url)
+        self.assertEqual(expected_status, resp.status_code)
+        as_json(resp)
+        self.assertIn("HEAD", resp['ALLOW'])
+        self.assertIn("OPTIONS", resp['ALLOW'])
+        self.assertIn("GET", resp['ALLOW'])
+
+        resp = self.client.head(url)
+        self.assertEqual(expected_status, resp.status_code)
+
+    def get_repositories(self, expected_status=HTTP_200_OK,
+                         skip_options_head_test=False):
+        """Get list of repositories"""
+        url = REPO_BASE
+        if not skip_options_head_test:
+            self.assert_options_head(url, expected_status=expected_status)
+        resp = self.client.get(url)
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == HTTP_200_OK:
+            return as_json(resp)
+
+    def create_repository(self, repo_dict=DEFAULT_REPO_DICT,
+                          expected_status=HTTP_201_CREATED,
+                          skip_assert=False):
+        """Helper function to create repository"""
+
+        resp = self.client.post(REPO_BASE, repo_dict)
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == HTTP_201_CREATED:
```

Comment:
nit:

This would be clearer as:

```python
if resp.status_code == HTTP_201_CREATED:
```

Since the `if` only cares if it's a 201, not about `expected_status`.